src/plugins/fileview/fileview.tsx

```diff
@@ -0,0 +1,523 @@
+        mobx.action(() => {
+            let searchText = this.tempSearchField.get();
+            this.props.model.handleSearch(searchText);
+        })();
+    }
+
+    renderSearchModal() {
+        let rendererModel = this.props.model;
+        return (
+            <If condition={rendererModel.search.get()}>
+                <Modal>
+                    <Modal.Header title="Search Filesystem" onClose={this.props.model.closeSearchModal}></Modal.Header>
+                    <TextField
+                        label="Search Field"
+                        autoFocus={true}
+                        value={this.tempSearchField.get()}
+                        onChange={(value) => {
+                            mobx.action(() => {
+                                this.tempSearchField.set(value);
+                            })();
+                        }}
+                        required={true}
+                        decoration={{
+                            endDecoration: (
+                                <InputDecoration>
+                                    <Tooltip
+                                        message={`The query that you would like to search in the filesystem from this directory`}
+                                        icon={<i className="fa-sharp fa-regular fa-circle-question" />}
+                                    >
+                                        <i className="fa-sharp fa-regular fa-circle-question" />
+                                    </Tooltip>
+                                </InputDecoration>
+                            ),
+                        }}
+                    />
+
+                    <Modal.Footer
+                        onCancel={this.props.model.closeSearchModal}
+                        onOk={this.handleSearchSubmit}
+                        okLabel="Ok"
+                    />
+                </Modal>
+            </If>
+        );
+    }
+
+    renderFile(file: any, index: number) {
+        let keyString = "file-" + index;
+        let downloadFileInputRef: React.RefObject<HTMLInputElement> = React.createRef();
+        const downloadButtonClicked = (event) => {
+            event.stopPropagation();
+            downloadFileInputRef.current.setAttribute("directory", "");
+            downloadFileInputRef.current.setAttribute("webkitdirectory", "");
+            downloadFileInputRef.current.click();
+        };
+        return (
+            <div
+                className={cn("file-container", {
+                    "dir-container": file.isdir,
+                })}
+                key={keyString}
+                onClick={(event) => this.props.model.fileWasClicked(event, file)}
+            >
+                {file.name}
+                <If condition={!file.isdir}>
+                    <div className="download-button" onClick={(event) => downloadButtonClicked(event)}>
+                        <i className="fa-sharp fa-solid fa-download"></i>
+                        <input
```

Comment:
Currently the file dialog on Mac says "upload". I couldn't find a way to change this online, does any one know how? 
It would be nice to be able to set a custom text like "upload" / "download", or just "open"